### DISCUSSION TITLE
Best practice sharing  Auth between instances

### DISCUSSION THREAD

**Discussion by tluyben** [supabase/supabase#2098] [Questions]
We are looking for a backend to create new microservice systems with and Supabase seems really good for many things, also as we already use Postgres. I have been playing around with examples etc only for now but I am wondering what the best way is to, for instance, share auth functionality (like api keys etc) with multiple instances. So we would have 1 Supabase which is auth and has custom api functionality for what we need. How can we share that so other Supabases take this one for auth but not other functionality? So that I don't have to struggle on all instances with custom auth which calls apis on the first one.

**Comment by silentworks** [comment]
I'm not sure I fully understand your question, but using the same API key provided by supabase for all instances of the `supabase-js` library would give you the same access to all its auth providers. i.e github, twitter, apple

**Reply by tluyben** [reply]
Thank you for the answer; my question is; if I have multiple supabase instances, how can I make sure they share the same user base, so there is only one user base across multiple instances of supabase. Does that make sense?

**Comment by ghost** [comment]
This is what I'm trying to do. I was wondering if it could be as simple as sharing the same JWT secret between the Supabase instances and keeping rows of the `auth.users` table synchronized.
